demos/free_surface/explicit_free_surface.py

```diff
@@ -0,0 +1,187 @@
+from gadopt import *
+from gadopt.utility import upward_normal
+from test_viscous_surface import run_benchmark
+
+
+class ExplicitFreeSurfaceModel:
+    # Test case from Section 3.1.1 of `An implicit free surface algorithm
+    # for geodynamical simulations', Kramer et al 2012.
+
+    name = "explicit"
+    bottom_free_surface = False
+
+    def __init__(self, dt_factor, nx=80, do_write=False, cartesian=True):
```

Comment:
As mentioned in the comment for `test_viscous_surface.py`, maybe this should also take kwargs, for unknowns that are passed to subclasses (e.g. `iterative_2d`):

```suggestion
    def __init__(self, dt_factor, nx=80, do_write=False, cartesian=True, **kwargs):
```

Reply:
done